modules/govuk_containers/manifests/tensorflow_serving.pp

```diff
@@ -0,0 +1,47 @@
+# == Class: govuk_containers::tensorflow_serving
+#
+# Install and run a dockerised Tensorflow server
+#
+# === Parameters
+#
+# [*image_name*]
+#   Docker image name to use for the container.
+#
+# [*image_version*]
+#   The docker image version to use.
+#
+# [*max_memory*]
+#   The maximum amount of memory Tensorflow Serving should allocate
+#
+class govuk_containers::tensorflow_serving(
+  $image_name = 'tensorflow/serving',
+  $image_version = 'latest',
+  $max_memory    = 1024,
+  $model_name = 'ltr'
+) {
+
+  ::docker::image { $image_name:
+    ensure    => 'present',
+    require   => Class['govuk_docker'],
+    image_tag => $image_version,
+  }
+
+  file { '/tensorflow-models':
+    ensure => directory,
+  } ->
+
+  ::docker::run { 'tensorflow/serving':
+    ports            => ['8501:8501'],
+    image            => $image_name,
+    require          => Docker::Image[$image_name],
+    extra_parameters => ['-P', '-v', '/tensorflow-models:/models'],
+    command          => "-m ${max_memory} -e MODEL_NAME=${model_name}",
```

Comment:
It might be that these can go into `extra_parameters`, I am not sure they are being passed yet. I get `MODEL_NAME=model` in the docker container.